app/src/main/java/com/github/h3lp3rs/h3lp/database/MockDatabase.kt

```diff
@@ -182,28 +179,29 @@ class MockDatabase : Database {
 
     @SuppressLint("RestrictedApi")
     override fun <T> addEventListener(
-        key : String?,
-        type : Class<T>,
-        onChildAdded : ((T) -> Unit)?,
-        onChildRemoved : (String) -> Unit
+        key: String?,
+        type: Class<T>,
+        onChildAdded: ((T) -> Unit)?,
+        onChildRemoved: (String) -> Unit
     ) {
-        val wrappedAction : () -> Unit = {
-            onChildAdded?.let {
-                val v : T =
-                    if (type == String::class.java || type == Int::class.java || type == Double::class.java || type == Boolean::class.java) {
-                        convertToCustomClass(db[key], type)
-                    } else {
-                        getObject(key!!, type).get()
+        synchronized(this) {
+            key?.let { k ->
+                val action: (List<T>) -> Unit = { list ->
+                    onChildAdded?.let { it(list.last()) }
+                }
+                addListListener(k, type, action)
+                if (concurrentLists.containsKey(k)) {
+                    // Reconstructing the entire list to be able to trigger it once for all n-1 entries
+                    val gson = Gson()
+                    val objectsList =
+                        concurrentLists.getOrDefault(key, emptyList())
+                            .map { gson.fromJson(it, type) }
+                            .dropLast(1)
+                    for (obj in objectsList) {
+                        onChildAdded?.let { it(obj) }

```

Comment:
The method was indeed totally incorrect, the solution is not very pretty but that's not your fault, it's better to only use the listListeners although not practical here.